### ISSUE TITLE
`stack path` outputs diagnostics to stdout

### ISSUE THREAD

**Issue by thomasvnoort** [commercialhaskell/stack#873]
When I run `stack path` while another stack instance is running, it outputs information about acquiring a lock to stdout. This makes it harder for build scripts to deal with the output of `stack path`, I'd expect such diagnostics to go to stderr. 

Also, adding `--verbosity SILENT` doesn't make a difference, but I'm not sure if that's an issue. If so, I'd be happy to create a separate issue for that.

Note that this issue is related to #872, if no locks are needed for `stack path` there might not be an issue here at all.

**Comment by snoyberg** [comment]
Can you provide the `stack --version` information? And can you test the latest master, which should resolve the issue?

**Comment by thomasvnoort** [comment]
You're right, this is no longer an issue with master from 3 days ago. I should've tried that version first, sorry about that.

Adding `--verbosity silent` still prints the diagnostics though, is that intentional? I've tested this with `Version 0.1.3.1, Git revision 1636b7cfb5d393298277f134dcfcb7208b18728d X86_64`.

**Comment by snoyberg** [comment]
I agree it would be better if it didn't, but I think there were technical reasons it didn't happen this way. @rrnewton do you remember those details?

**Comment by rrnewton** [comment]
I think this was just a case of me coming in blind and not knowing enough of the infrastructure.  If `--verbosity silent` turns of all stderr/stdout output, then it should definitely turn off this message too.

The rationale for the message of course is explaining to the user why its hanging there, and giving them the option of finding and deleting the lock in the worst, worst case of us messing up somehow.

**Comment by snoyberg** [comment]
Do you feel comfortable taking a stab at making the change? If you need
guidance, let us know, I'm sure someone from the team can have a look with
you.

**Comment by rrnewton** [comment]
Having a look now....

It is currently outputting to `stderr` btw, not `stdout`.  I see many other commands that use `hPutStrLn stderr`, but I now see those are only when bailing out with an error.  Ok, I found `LogLevel` and put in a test to see if it is `LevelOther "silent"`.  With some simple testing, this seems to work.  I made a pull request for it.

However, I notice that other output actually disobeys `--verbosity=silent`.  For example, the following test doesn't print "grabbing lock" messages, but it does print other chatter about the index cache and progress:

```bash
> ($STACK --verbosity=silent build &); $STACK --verbosity=silent build
Populated index cache.
Progress: 0/3
```
